entity-data-service-rx-client/src/main/java/org/hypertrace/entity/data/service/rxclient/EntityCacheKey.java

```diff
@@ -0,0 +1,62 @@
+package org.hypertrace.entity.data.service.rxclient;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import org.hypertrace.core.grpcutils.client.rx.GrpcRxExecutionContext;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.entity.data.service.v1.Entity;
+
+class EntityCacheKey {
+
+  private static final String DEFAULT_TENANT_ID = "default";
+
+  static EntityCacheKey entityInCurrentContext(Entity inputEntity) {
+    return new EntityCacheKey(
+        requireNonNull(RequestContext.CURRENT.get()), requireNonNull(inputEntity));
+  }
+
+  private final Entity inputEntity;
+  private final String tenantId;
+  private final GrpcRxExecutionContext executionContext;
+
+  protected EntityCacheKey(@Nonnull RequestContext requestContext, @Nonnull Entity inputEntity) {
+    requireNonNull(inputEntity.getEntityId());
+    requireNonNull(inputEntity.getEntityType());
+    this.executionContext = GrpcRxExecutionContext.forContext(requestContext);
+    this.tenantId = requestContext.getTenantId().orElse(DEFAULT_TENANT_ID);
```

Comment:
good or bad idea to have a default?

Reply:
So my thinking here was that it shouldn't happen, but if it were to, it's irrelevant to the role of this object as a cache key - i'd rather have the downstream call decide whether it actually needs that tenant id or not. I could definitely be convinced otherwise though.